ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/MaxMinMarginFiller.java

```diff
@@ -49,11 +52,18 @@ public void fill(LinearProblem linearProblem, FlowResult flowResult, Sensitivity
 
         // build constraints
         buildMinimumMarginConstraints(linearProblem, validFlowCnecs);
+        if (costOptimization) {
+            forceMinMarginToBeNegative(linearProblem);
```

Comment:
add a comment to explain we do this to penalize negative margins without rewarding positive margins